### PR TITLE
[WIP][ENH] Add deselect all button and auto commit of filter results.

### LINKED ISSUE #156: Marker Genes: output under no selection of genes
##### Single Cell version
0.9.1

##### Expected behavior
Under no selection of genes, Marker Genes should output the list of genes currently displayed in the preview window. The output should change under any change of filtering. That is, when changing the entry in the filter box, if this changes the list of genes, the change should be reflected on the output. (if the list of genes does not change when changing the filter box, there is no need to refresh the output).

##### Actual behavior
Under no selection, the widget outputs all the genes for a selected organism, and the output does not change with the change of filtering.

##### Steps to reproduce the behavior
screen shot 2019-02-13 at 07 48 48

### LINKED ISSUE #158: [ENH] Number of selected marker genes and clear selection in marker genes widget
Marker genes widget does not have an easy option to deselect all marker genes. Today at the meeting we have talked about button `Clear selection` which will be placed below the table. 

Also, there would be nice to have information about a number of all genes and the number of selected genes in the status bar - bottom of the widget.

### PR DESCRIPTION
**Pull request by NejcDebevec**
##### Issue

Fixes #158 
Fixes #156 

##### Includes
- [X] Code changes
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: orangecontrib/bioinformatics/widgets/OWMarkerGenes.py [hunk 1]
```diff
@@ -370,6 +395,15 @@ def _filter_table(self):
...
         model.update_model(str(self.filter_text))
+        # if self.filter_commit:
+        self.commit()
+        self.update_data_info()
+
+    def update_data_info(self):
+        filtered_data = len(self.view.model().source)
+        all_data = len(self.view.model()._data)
+        self.info.set_input_summary(f"Shown : {str(filtered_data)}/{str(all_data)}")
+        self.info.set_output_summary("Selected: "+str(len(self.selected_genes)))
```
Comment (JakaKokosar) [inline orangecontrib/bioinformatics/widgets/OWMarkerGenes.py]:
```suggestion
        self.info.set_output_summary(f"Selected: {str(len(self.selected_genes))}")
```

File: orangecontrib/bioinformatics/widgets/OWMarkerGenes.py [hunk 2]
```diff
@@ -84,12 +84,14 @@ def __init__(self, data, parent):
             partial(defaultdict, partial(defaultdict, lambda: None)),
         )(self._roleData)
         self.set_column_links()
+        self.source = data
```
Comment (JakaKokosar) [inline orangecontrib/bioinformatics/widgets/OWMarkerGenes.py]: Why is this necessary (source attribute should only be set by wrap method in SearchableTableModel)?
I think this can be safely removed.

File: orangecontrib/bioinformatics/widgets/OWMarkerGenes.py [hunk 3]
```diff
@@ -370,6 +395,15 @@ def _filter_table(self):
         model = self.view.model()
         assert isinstance(model, SearchableTableModel)
         model.update_model(str(self.filter_text))
+        # if self.filter_commit:
```
Comment (JakaKokosar) [inline orangecontrib/bioinformatics/widgets/OWMarkerGenes.py]: This can be removed

File: orangecontrib/bioinformatics/widgets/OWMarkerGenes.py [hunk 4]
```diff
@@ -370,6 +395,15 @@ def _filter_table(self):
...
+        self.commit()
+        self.update_data_info()
+
+    def update_data_info(self):
+        filtered_data = len(self.view.model().source)
+        all_data = len(self.view.model()._data)
```
Comment (JakaKokosar) [inline orangecontrib/bioinformatics/widgets/OWMarkerGenes.py]: access to protected member _data of a class.

File: orangecontrib/bioinformatics/widgets/OWMarkerGenes.py (hunk 2 above)
Comment (NejcDebevec) [reply orangecontrib/bioinformatics/widgets/OWMarkerGenes.py]: I removed this line in last commit.

File: orangecontrib/bioinformatics/widgets/OWMarkerGenes.py (hunk 3 above)
Comment (NejcDebevec) [reply orangecontrib/bioinformatics/widgets/OWMarkerGenes.py]: I removed it.

File: orangecontrib/bioinformatics/widgets/OWMarkerGenes.py (hunk 1 above)
Comment (NejcDebevec) [reply orangecontrib/bioinformatics/widgets/OWMarkerGenes.py]: I fixed this in last commit

File: orangecontrib/bioinformatics/widgets/OWMarkerGenes.py [hunk 5]
```diff
@@ -370,6 +399,14 @@ def _filter_table(self):
...
         assert isinstance(model, SearchableTableModel)
         model.update_model(str(self.filter_text))
+        self.commit()
+        self.update_data_info()
+
+    def update_data_info(self):
+        filtered_data = len(self.view.model().source)
+        all_data = len(self.view.model()._data)
+        self.info.set_input_summary(f"Shown : {str(filtered_data)}/{str(all_data)}")
+        self.info.set_output_summary(f"Selected: {str(len(self.selected_genes))}")
```
Comment (PrimozGodec) [inline orangecontrib/bioinformatics/widgets/OWMarkerGenes.py]: Those two lines are longer than 79 characters. As far I know we are still keeping the line lengths at maximum 79 as it is declared PEP8.

**Comment by PrimozGodec** [comment]
I have also found the other issue: There is an inconsistency of selecting all genes when the data source changes. When auto-commit is on all genes are selected on source change (database or organism) but when auto-commit is of selection does not happen and all genes are not selected.

Everything else looks good.

**Comment by PrimozGodec** [comment]
The widget looks great now. There is only one minor change required - When I click to Clear selection number of selected genes in the status bar does not change. When auto-commit is off a number of selected genes at the output is still changing. It should only change on commit.

**Comment by JakaKokosar** [comment]
Closing this PR. More info: https://github.com/biolab/orange3-bioinformatics/issues/180
